packages/provider/test/null_safe/selector_test.dart

```diff
@@ -334,6 +334,37 @@ void main() {
     expect(find.text('24'), findsOneWidget);
   });
 
+  testWidgets('rebuild when inherited widget changes', (tester) async {
+    when(selector(any)).thenReturn(42);
+
+    mockBuilder((_, value, ___) {
+      return Text(value.toString(), textDirection: TextDirection.ltr);
+    });
+
+    final selectorWidget = Selector0<int>(
+      selector: selector,
+      builder: (context, _, __) => builder(context, DummyInheritedWidget.of(context).data, null),
+    );
+
+    await tester.pumpWidget(
+      DummyInheritedWidget(
+        data: 1,
+        child: selectorWidget,
+      ),
+    );
+
+    verify(builder(argThat(isNotNull), 1, null)).called(1);
```

Comment:
Rather than mocking/spying `build`, could you use `expect(find.text('1'), findsOneWidget); expect(find.text('0'), findNothing);`?

Reply:
Done